[Lee Wei Chong Stefan] iP

rexcyrio opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them." -- David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

rexcyrio committed:
Level-1: echo user input

rexcyrio committed:
Level-2: maintain a list of user tasks

rexcyrio committed:
Level-3: user can mark tasks as complete

rexcyrio committed:
Level-4: new types of tasks

Introduced normal to-do tasks, tasks with deadlines and event tasks

rexcyrio committed:
A-TextUiTesting: testing app

rexcyrio committed:
Level-5: handling errors using Exceptions

rexcyrio committed:
Level-6: user can delete tasks

rexcyrio committed:
A-Enums: user actions are enumerable

rexcyrio committed:
Level-7: automatically save user tasks to disk

rexcyrio committed:
Level-8: use proper dates for `DeadlineTask`

Also add new `dueon` command

rexcyrio committed:
Merge branch 'branch-Level-7'

rexcyrio committed:
Merge branch 'branch-Level-8'

rexcyrio committed:
A-MoreOOP: refactor code

Use `Ui`, `Storage`, `Parser`, `TaskList` and `*Command` classes

rexcyrio committed:
A-Packages: organise code into packages

rexcyrio committed:
A-Gradle: fix `mainClassName`

rexcyrio committed:
Merge branch 'branch-A-Gradle'

rexcyrio committed:
A-JUnit: test code using JUnit

rexcyrio committed:
A-JavaDoc: add JavaDoc comments

rexcyrio committed:
A-CodingStandard: remove unnecessary `this`

rexcyrio committed:
Level-9: user can search for tasks by name

rexcyrio committed:
Merge branch 'branch-A-JavaDoc'

rexcyrio committed:
Merge branch 'branch-A-CodingStandard'

rexcyrio committed:
Merge branch 'branch-Level-9'

sumhungyee commented on src/main/java/duke/utilities/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+     * Deletes the task at the specified position from the task list.
+     *
+     * @param index The index of the task to be deleted.
+     * @return The task that was deleted.
```
Great job formatting your javadocs!

sumhungyee commented on src/main/java/duke/utilities/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+        return task;
+    }
+
+    /**
+     * Returns all tasks that are due on the specified date.
+     *
+     * @param dueOnDate The date on which the tasks are due on.
+     * @return A list of tasks that are due on {@code dueOnDate}
+     */
```
Great job in general for capitalising and punctuating all your javadoc comments!

sumhungyee commented on src/test/java/duke/commands/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    @Test
+    public void execute_noTaskNameNoByNoDeadline_DukeInvalidDeadlineCommandExceptionThrown() {
```
Great job for remembering to format your test functions as well!

sumhungyee commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /**
+     * Marks the task as done
```
Perhaps a fullstop?

sumhungyee commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    /**
+     * Unmarks the task as done
```
Perhaps a fullstop here as well!

sumhungyee reviewed (commented):
Great job! Your formatting is impeccable! Well done for sticking to the appropriate coding standards! 🦾

rexcyrio committed:
A-CheckStyle: use `checkstyle.xml`

rexcyrio committed:
Missing `private` access modifier

rexcyrio committed:
Level-10: new GUI for `Duke`

clevon-w commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,81 @@
+        }
+    }
+
+    public String getResponse(String line) {
```
you missed out JavaDocs here 😉

clevon-w commented on src/test/java/duke/commands/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,99 @@
+            DeadlineCommand deadlineCommand = new DeadlineCommand(tokens);
+            deadlineCommand.execute(taskList, ui, storage);
+        });
+    }
```
Great job testing exceptions thrown as well!

clevon-w reviewed (commented):
Really great job Stefan! Your attention to detail is amazing! Looking through your code just showed me that I have a lot of room for improvement!

rexcyrio committed:
Merge branch 'branch-A-CheckStyle'

rexcyrio committed:
Merge branch 'branch-Level-10'

rexcyrio committed:
A-Assertions: use and enable assertions

Assert that the `formatNumberOfTasksAsString` function only receives
non-negative integer arguments

rexcyrio committed:
A-CodeQuality: abstract out `parseTask` method

Each task type will now have their own parsing function

rexcyrio committed:
Merge pull request #1 from rexcyrio/branch-A-Assertions

A-Assertions: use and enable assertions

rexcyrio committed:
Merge branch 'master' into branch-A-CodeQuality

rexcyrio committed:
Merge pull request #2 from rexcyrio/branch-A-CodeQuality

Branch a code quality

rexcyrio committed:
A-CI: use GitHub actions

rexcyrio committed:
All files should belong to a package

rexcyrio committed:
Disambiguate the Duke GUI from Duke itself

rexcyrio committed:
C-Undo: user can undo previous commands

rexcyrio committed:
Merge branch 'branch-BCD-Extension'

rexcyrio committed:
A-UserGuide: table of contents & screenshots

rexcyrio committed:
Merge branch 'branch-A-UserGuide'

rexcyrio committed:
Bug fix: show GUI error if `duke.txt` is corrupted
